### PR TITLE
Clone of PR support Sendgrid API v3

### PR DESCRIPTION
**Pull request by andriisoldatenko**
Clone of PR support Sendgrid API v3 and Python3
https://github.com/elbuo8/sendgrid-django/pull/23
because of any activities.

### CODE REVIEW

**Comment by andriisoldatenko** [comment]
@elbuo8 please review.

File: sgbackend/mail.py [hunk 1]
```diff
@@ -51,67 +44,57 @@ def send_messages(self, emails):
...
-                count += 1
-            except sendgrid.SendGridClientError:
-                if not self.fail_silently:
-                    raise
-            except sendgrid.SendGridServerError:
-                if not self.fail_silently:
-                    raise
+            mail = self._build_sg_mail(email)
+            response = self.sg.client.mail.send.post(request_body=mail)
+            count += 1
```
Comment (elbuo8) [inline sgbackend/mail.py]: New lib doesn't raise errors and it's up to the user to inspect the `status_code` field? If so, we should check for `OK` to update this counter.

File: sgbackend/mail.py [hunk 2]
```diff
@@ -51,67 +44,57 @@ def send_messages(self, emails):
...
-                    mail.set_html(alt[0])
+                    mail.add_content(Content(alt[1], alt[0]))
         elif email.content_subtype == "html":
-            mail.set_html(email.body)
-            mail.set_text("")
+            mail.contents = []
+            mail.add_content(Content("text/plain", ' '))
+            mail.add_content(Content("text/html", email.body))
+
+        if hasattr(email, 'sendgrid_categories'):
```
Comment (elbuo8) [inline sgbackend/mail.py]: Why the need for the `sendgrid_` prefix?

File: sgbackend/mail.py [hunk 3]
```diff
@@ -51,67 +44,57 @@ def send_messages(self, emails):
...
+            personalization.add_cc(Email(e))
+        for e in email.bcc:
+            personalization.add_bcc(Email(e))
+        personalization.set_subject(email.subject)
+        mail.add_content(Content("text/plain", email.body))
         if isinstance(email, EmailMultiAlternatives):
             for alt in email.alternatives:
                 if alt[1] == "text/html":
-                    mail.set_html(alt[0])
+                    mail.add_content(Content(alt[1], alt[0]))
```
Comment (elbuo8) [inline sgbackend/mail.py]: 👍

File: sgbackend/mail.py (hunk 1 above)
Comment (thinkingserious) [inline sgbackend/mail.py]: Yes, for now you need to catch the errors like this: https://github.com/sendgrid/sendgrid-python/blob/master/TROUBLESHOOTING.md#error-messages

**Comment by andriisoldatenko** [comment]
@elbuo8 Can you please review again?

**Comment by elbuo8** [comment]
@andriisoldatenko left last comment. After that I think this will be 💪

**Comment by andriisoldatenko** [comment]
@elbuo8 Sorry, which one?

**Comment by elbuo8** [comment]
@andriisoldatenko sorry, forgot to submit it. Should be visible now.

File: sgbackend/mail.py [hunk 4]
```diff
@@ -51,67 +51,63 @@ def send_messages(self, emails):
...
                 count += 1
-            except sendgrid.SendGridClientError:
-                if not self.fail_silently:
-                    raise
-            except sendgrid.SendGridServerError:
-                if not self.fail_silently:
-                    raise
-
+            except urllib.error.HTTPError as e:
+                logger.error(e.read())
```
Comment (andriisoldatenko) [inline sgbackend/mail.py]: Fixed.

**Pull request merged**